src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -304,16 +305,38 @@ private void BuildPreprocessorStateChangeMap()
 
         #region Factories
 
+        // The overload that has more parameters is itself obsolete, as an intentional break to allow future
+        // expansion
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
+
+        /// <summary>
+        /// Creates a new syntax tree from a syntax node.
+        /// </summary>
+        public static SyntaxTree Create(CSharpSyntaxNode root, CSharpParseOptions? options = null, string path = "", Encoding? encoding = null)
+        {
+#pragma warning disable CS0618 // We are calling into the obsolete member as that's the one that still does the real work
```

Comment:
Why do it in that order? I'd rather have the obsolete one call into this method.

Reply:
The diagnosticOptions API isn't _entirely_ broken in this change: the value passed is still handed to the tree, although it's not used in the regular command line compiler anymore. That was a decision @agocke intentionally made for this PR and I'm preserving it.